validate/validate.go

```diff
@@ -0,0 +1,574 @@
+				msgs = append(msgs, fmt.Sprintf("Unsupported mount type %q", mount.Type))
+			}
+		}
+	}
+
+	return
+}
+
+//Linux only
+func (v *Validator) CheckLinux() (msgs []string) {
+	logrus.Debugf("check linux")
+
+	utsExists := false
+	ipcExists := false
+	mountExists := false
+	netExists := false
+	userExists := false
+
+	for index := 0; index < len(v.spec.Linux.Namespaces); index++ {
+		if !namespaceValid(v.spec.Linux.Namespaces[index]) {
+			msgs = append(msgs, fmt.Sprintf("namespace %v is invalid.", v.spec.Linux.Namespaces[index]))
+		} else if len(v.spec.Linux.Namespaces[index].Path) == 0 {
+			if v.spec.Linux.Namespaces[index].Type == rspec.UTSNamespace {
+				utsExists = true
+			} else if v.spec.Linux.Namespaces[index].Type == rspec.IPCNamespace {
+				ipcExists = true
+			} else if v.spec.Linux.Namespaces[index].Type == rspec.NetworkNamespace {
+				netExists = true
+			} else if v.spec.Linux.Namespaces[index].Type == rspec.MountNamespace {
+				mountExists = true
+			} else if v.spec.Linux.Namespaces[index].Type == rspec.UserNamespace {
+				userExists = true
+			}
+		}
+	}
+
+	if (len(v.spec.Linux.UIDMappings) > 0 || len(v.spec.Linux.GIDMappings) > 0) && !userExists {
+		msgs = append(msgs, "UID/GID mappings requires a new User namespace to be specified as well")
+	} else if len(v.spec.Linux.UIDMappings) > 5 {
+		msgs = append(msgs, "Only 5 UID mappings are allowed (linux kernel restriction).")
+	} else if len(v.spec.Linux.GIDMappings) > 5 {
+		msgs = append(msgs, "Only 5 GID mappings are allowed (linux kernel restriction).")
+	}
+
+	for k := range v.spec.Linux.Sysctl {
+		if strings.HasPrefix(k, "net.") && !netExists {
+			msgs = append(msgs, fmt.Sprintf("Sysctl %v requires a new Network namespace to be specified as well", k))
+		}
+		if strings.HasPrefix(k, "fs.mqueue.") {
+			if !mountExists || !ipcExists {
+				msgs = append(msgs, fmt.Sprintf("Sysctl %v requires a new IPC namespace and Mount namespace to be specified as well", k))
+			}
+		}
+	}
+
+	if v.spec.Platform.OS == "linux" && !utsExists && v.spec.Hostname != "" {
+		msgs = append(msgs, fmt.Sprintf("On Linux, hostname requires a new UTS namespace to be specified as well"))
+	}
+
+	for index := 0; index < len(v.spec.Linux.Devices); index++ {
+		if !deviceValid(v.spec.Linux.Devices[index]) {
+			msgs = append(msgs, fmt.Sprintf("device %v is invalid.", v.spec.Linux.Devices[index]))
+		}
+	}
+
+	if v.spec.Linux.Resources != nil {
+		ms := v.checkLinuxResources()
+		msgs = append(msgs, ms...)
+	}
+
+	if v.spec.Linux.Seccomp != nil {
+		ms := v.checkSeccomp()
+		msgs = append(msgs, ms...)
+	}
+
+	switch v.spec.Linux.RootfsPropagation {
+	case "":
+	case "private":
+	case "rprivate":
+	case "slave":
+	case "rslave":
+	case "shared":
+	case "rshared":
+	default:
+		msgs = append(msgs, "rootfsPropagation must be empty or one of \"private|rprivate|slave|rslave|shared|rshared\"")
+	}
+
+	return
+}
+
+func (v *Validator) checkLinuxResources() (msgs []string) {
```

Comment:
This might as well be public, so folks who want to can check it without having to run all of `CheckLinux`.